sktime/transformers/series_as_features/signature_based/_rescaling.py

```diff
@@ -0,0 +1,124 @@
+"""
+rescaling.py
+=========================
+Signature rescaling methods. This implements the pre- and post- signature
+rescaling methods along with generic scaling methods along feature
+dimensions of 3D tensors.
+
+Code for `rescale_path` and `rescale_signature` written by Patrick Kidger.
+"""
+import math
+import torch
+from sklearn.preprocessing import StandardScaler, MinMaxScaler, MaxAbsScaler, \
+    FunctionTransformer
+from sktime.transformers.series_as_features.base import \
+    BaseSeriesAsFeaturesTransformer
+import signatory
+
+
+class TrickScaler(BaseSeriesAsFeaturesTransformer):
+    """Tricks an sklearn scaler so that it uses the correct dimensions.
+
+    This class was created out of a desire to use sklearn scaling functionality
+    on 3D tensors. Sklearn operates on a tensor of shape [N, C] and
+    normalises along the channel dimensions. To make this functionality work on
+    tensors of shape [N, L, C] we simply first stack the first two dimensions
+    to get shape [N * L, C], apply a scaling function and finally stack back
+    to shape [N, L, C].
+
+    Parameters
+    ----------
+    scaling : str, Scaling method, one of ['stdsc', 'maxabs', 'minmax', None].
+
+    # TODO allow scaling to be passed as an sklearn transformer.
+    """
+    def __init__(self, scaling):
+        self.scaling = scaling
+        if scaling == 'stdsc':
+            self.scaler = StandardScaler()
+        elif scaling == 'maxabs':
+            self.scaler = MaxAbsScaler()
+        elif scaling == 'minmax':
+            self.scaler = MinMaxScaler()
+        elif scaling is None:
+            self.scaler = FunctionTransformer(func=None)
+        else:
+            raise ValueError('scaling param {} not recognised.'
+                             ''.format(scaling))
+
+    def _trick(self, X):
+        return X.reshape(-1, X.shape[2])
+
+    def _untrick(self, X, shape):
+        return X.reshape(shape)
+
+    def fit(self, X, y=None):
+        self.scaler.fit(self._trick(X), y)
+        self._is_fitted = True
+        return self
+
+    def transform(self, X):
+        X_tfm = self.scaler.transform(self._trick(X))
```

Comment:
input checks and is fitted checks are missing

Reply:
This is only ever called with a tensor and not with a sktime type dataframe input therefore it is not expected to pass said input checks.